### PR TITLE
feature(menus): you can now register togglable menu items from php

### LINKED ISSUE #8158: add a way to add toggle menu items from php / output/url states
We currently have this js:
'elgg.ui.registerTogglableMenuItems('feature', 'unfeature');'

can't we come up with a way to autodetect those with a generic js function? or register the menu items as toggleable from php?

This could prevent adding js files or have inline js for just toggleable menu items.

Any preference?

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #8158 
- [x] write/update docs

### CODE REVIEW

File: engine/lib/friends.php [hunk 1]
```diff
@@ -87,6 +87,7 @@ function _elgg_friends_setup_user_hover_menu($hook, $type, $return, $params) {
 			// Always emit both to make it super easy to toggle with ajax
 			$return[] = \ElggMenuItem::factory(array(
 				'name' => 'remove_friend',
+				'data-toggle' => 'add_friend',
```
Comment (hypeJunction) [inline engine/lib/friends.php]: Let's use `data-toggleable-item` or something of that nature. We may want to reserve `data-toggle` for `rel="toggle"` refactor. Or use class.

**Comment by hypeJunction** [comment]
I would say let's deprecate tobbleable menu api from `elgg.ui.*`, add `elgg/menu` AMD and inline it in `elgg.js` while at it. I don't feel like we should be adding more API to `elgg.ui.*`. There is enough headache deprecating what we have and moving to AMD.

**Comment by hypeJunction** [comment]
Also, I don't feel like deprecating client-side API in favour of server-side API is such a good idea. What if you want to just register two toggleable items in JS? Do you have to add classes/data attributes to them?

**Comment by jdalsem** [comment]
I prefer not to move to an AMD as part of this 'small' PR. There is a lot of menu related stuff that should be moved around and i think we should think about restructuring that a bit more...



> What if you want to just register two toggleable items in JS?

Currently you can keep using the deprecated function, but we really need to deprecate it, as it is prone to error. If you have multiple menu items with the same menu item name, but in a different menu this current function is prone to error as it binds to all menu items with the same name.



> Do you have to add data attributes to them

That is the new situation if you do not want to use the deprecated function and cannot use the php alternative approach

File: engine/lib/friends.php (hunk 1 above)
Comment (jdalsem) [inline engine/lib/friends.php]: i went for `toggle` in the factory.... in elgg_view_menu_item this gets translated to `data-toggleable-item`

**Comment by jdalsem** [comment]
replaced with #10478

**Pull request closed without merging**